File: EaterSpring/src/main/kotlin/xyz/purema/eater/dtos/form/RegisterForm.kt
package xyz.purema.eater.dtos.form

import org.hibernate.validator.constraints.Length
import javax.validation.constraints.NotEmpty

class RegisterForm {
    @field:NotEmpty(message = "Please enter user id")
    var userId: String? = null

    @field:NotEmpty(message = "Please enter name")
    var name: String? = null

    @field:NotEmpty(message = "Please enter password")
    @field:Length(min = 6, message = "Password must have minimum of 6 characters")
    var password: String? = null
}

File: EaterNet/Eater/Dtos/Form/RegisterForm.cs
using System.ComponentModel.DataAnnotations;

namespace Eater.Dtos.Form
{
    public class RegisterForm
    {
        [Required(ErrorMessage = "Please enter user id")]
        public string UserId { get; set; }

        [Required(ErrorMessage = "Please enter name")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Please enter password")]
        [MinLength(6, ErrorMessage = "Password must have minimum of 6 characters")]
        public string Password { get; set; }

        public RegisterForm(string userId, string name, string password)
        {
            UserId = userId;
            Name = name;
            Password = password;
        }
    }
}
